### PR TITLE
Simplify check for writing errors to ErrorOutput

### PR DESCRIPTION
**Pull request by lancoLiu**

### CODE REVIEW

File: zapcore/entry.go [hunk 1]
```diff
@@ -219,11 +219,9 @@ func (ce *CheckedEntry) Write(fields ...Field) {
 	for i := range ce.cores {
 		err = multierr.Append(err, ce.cores[i].Write(ce.Entry, fields))
 	}
-	if ce.ErrorOutput != nil {
-		if err != nil {
-			fmt.Fprintf(ce.ErrorOutput, "%v write error: %v\n", ce.Time, err)
-			ce.ErrorOutput.Sync()
-		}
+	if ce.ErrorOutput != nil && err != nil {
```
Comment (prashantv) [inline zapcore/entry.go:222]:
```suggestion
	if err != nil && ce.ErrorOutput != nil {
```

**Review by prashantv — APPROVED** [review]
Thanks for the simplification!

File: zapcore/entry.go (hunk 1 above)
Comment (prashantv) [reply zapcore/entry.go:222]: This may be a little clearer logically -- if there's an error, *then* we care about whether we have an `ErrorOutput`.

Comment (lancoLiu) [reply zapcore/entry.go]: thanks. I agree with you.

**Comment by prashantv** [comment]
Thanks for the contribution @lancoLiu

**Pull request merged**